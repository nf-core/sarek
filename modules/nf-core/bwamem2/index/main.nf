process BWAMEM2_INDEX {
    tag "$fasta"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/bwa-mem2:2.2.1--he513fc3_0' :
        'biocontainers/bwa-mem2:2.2.1--he513fc3_0' }"

    input:
    tuple val(meta), path(fasta)

    output:
    tuple val(meta), path("bwamem2"), emit: index
    path "versions.yml"             , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${fasta}"
    def args = task.ext.args ?: ''
    """
    mkdir bwamem2
    bwa-mem2 \\
        index \\
        $args \\
        $fasta -p bwamem2/${prefix}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bwamem2: \$(echo \$(bwa-mem2 version 2>&1) | sed 's/.* //')
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${fasta}"

    """
    mkdir bwamem2
    touch bwamem2/${prefix}.0123
    touch bwamem2/${prefix}.ann
    touch bwamem2/${prefix}.pac
    touch bwamem2/${prefix}.amb
    touch bwamem2/${prefix}.bwt.2bit.64

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bwamem2: \$(echo \$(bwa-mem2 version 2>&1) | sed 's/.* //')
    END_VERSIONS
    """
}
