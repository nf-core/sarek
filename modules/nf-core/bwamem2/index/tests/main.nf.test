nextflow_process {

    name "Test Process BWAMEM2_INDEX"
    tag "modules_nfcore"
    tag "modules"
    tag "bwamem2"
    tag "bwamem2/index"
    script "../main.nf"
    process "BWAMEM2_INDEX"

    test("BWAMEM2 index") {

        when {
            process {
                """
                input[0] = [
                    [id: 'test'],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
