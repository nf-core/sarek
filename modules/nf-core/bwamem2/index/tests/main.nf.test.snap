{
    "BWAMEM2 index": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "genome.fasta.0123:md5,b02870de80106104abcb03cd9463e7d8",
                            "genome.fasta.amb:md5,3a68b8b2287e07dd3f5f95f4344ba76e",
                            "genome.fasta.ann:md5,c32e11f6c859f166c7525a9c1d583567",
                            "genome.fasta.bwt.2bit.64:md5,d097a1b82dee375d41a1ea69895a9216",
                            "genome.fasta.pac:md5,983e3d2cd6f36e2546e6d25a0da78d66"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,9ffd13d12e7108ed15c58566bc4717d6"
                ],
                "index": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "genome.fasta.0123:md5,b02870de80106104abcb03cd9463e7d8",
                            "genome.fasta.amb:md5,3a68b8b2287e07dd3f5f95f4344ba76e",
                            "genome.fasta.ann:md5,c32e11f6c859f166c7525a9c1d583567",
                            "genome.fasta.bwt.2bit.64:md5,d097a1b82dee375d41a1ea69895a9216",
                            "genome.fasta.pac:md5,983e3d2cd6f36e2546e6d25a0da78d66"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,9ffd13d12e7108ed15c58566bc4717d6"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-03-18T12:59:39.132616"
    }
}